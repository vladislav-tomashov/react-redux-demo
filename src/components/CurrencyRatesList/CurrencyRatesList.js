import React from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import moment from "moment";
import { getCurrencyRatesData } from "../../store/selectors/currencyRatesSelectors";
import CurrencyRatesItem from "../CurrencyRatesItem/CurrencyRatesItem";
import "./CurrencyRatesList.scss";

const CurrencyRatesList = ({ date, base, rates } = {}) => {
  const content = !date ? (
    <span>Please press [Load rates] button to get currency rates</span>
  ) : (
    <React.Fragment>
      <div className="CurrencyRates-heading">Currency rates</div>
      <div className="CurrencyRates-info">
        <span>Currency rates for:</span>
        <span>{moment(date).format("LL")}</span>
      </div>
      <div className="CurrencyRates-info">
        <span>Base currency:</span>
        <span>{base}</span>
      </div>
      <div className="CurrencyRates-header">
        <div>Currency</div>
        <div>Rate</div>
      </div>
      <div className="CurrencyRates-body">
        {Object.entries(rates).map(([currency, rate]) => {
          return <CurrencyRatesItem key={currency} {...{ currency, rate }} />;
        })}
      </div>
    </React.Fragment>
  );
  return <div className="CurrencyRates">{content}</div>;
};

CurrencyRatesList.propTypes = {
  rates: PropTypes.objectOf(PropTypes.number),
  date: PropTypes.instanceOf(Date),
  base: PropTypes.string
};

const mapStateToProps = (state /*, ownProps*/) => {
  return getCurrencyRatesData(state);
};

const ConnectedCurrencyRatesList = connect(mapStateToProps)(CurrencyRatesList);

export { ConnectedCurrencyRatesList as default, CurrencyRatesList };
