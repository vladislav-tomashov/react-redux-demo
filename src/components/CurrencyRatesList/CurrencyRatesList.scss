@import "../../styles/settings";

.CurrencyRates {
  margin: 0 auto;
  padding: 0 $m-size;
}

.CurrencyRates-heading {
  font-size: $font-size-xlarge;
  font-weight: 600;
  color: $light-blue;
}

.CurrencyRates-info {
  display: flex;
  justify-content: space-between;
  white-space: nowrap;
  text-align: left;
}

.CurrencyRates-info > * {
  padding: 0 $s-size;
}

.CurrencyRates-header {
  margin-top: $m-size;
  background: $light-blue;
  color: $black;
  display: flex;
  justify-content: space-between;
  padding: 0 $s-size;
  font-weight: 600;
}
.CurrencyRates-body {
  font-size: $font-size-small;
}
