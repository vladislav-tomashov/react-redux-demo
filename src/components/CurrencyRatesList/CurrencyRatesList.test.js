import React from "react";
import { shallow } from "enzyme";
import { CurrencyRatesList } from "./CurrencyRatesList";

let wrapper;

describe("CurrencyRatesList component", () => {
  beforeEach(() => {
    wrapper = shallow(<CurrencyRatesList />);
  });

  test("should render CurrencyRatesList correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
