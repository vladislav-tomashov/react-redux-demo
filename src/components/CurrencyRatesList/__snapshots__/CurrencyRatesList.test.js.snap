// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencyRatesList component should render CurrencyRatesList correctly 1`] = `
<div
  className="CurrencyRates"
>
  <span>
    Please press [Load rates] button to get currency rates
  </span>
</div>
`;
